### PR TITLE
feat:add Bidistreaming Api

### PR DESCRIPTION
**Pull request by HHoflittlefish777**
### Motivation:

Explain the context, and why you're making that change.
To make others understand what is the problem you're trying to solve.

### Modification:

Describe the idea and modifications you've done.

### Result:

Fixes 

If there is no issue then describe the changes introduced by this PR.

### CODE REVIEW

File: jraft-core/src/main/java/com/alipay/sofa/jraft/rpc/impl/BoltRpcClient.java [hunk 1]
```diff
@@ -119,6 +122,12 @@ public void invokeAsync(final Endpoint endpoint, final Object request, final Inv
         }
     }
 
+    @Override
+    public StreamObserver<Message> invokeBidiStreaming(Endpoint endpoint, Object request, InvokeContext ctx,
+                                                       InvokeCallback callback, long timeoutMs) {
+        return null;
```
Comment (fengjiachun) [inline jraft-core/src/main/java/com/alipay/sofa/jraft/rpc/impl/BoltRpcClient.java:128]: 可以 throw 一个 UnsupportedException 之类的 error

File: jraft-core/src/main/java/com/alipay/sofa/jraft/rpc/impl/BoltRpcClient.java [hunk 2]
```diff
@@ -119,6 +122,12 @@ public void invokeAsync(final Endpoint endpoint, final Object request, final Inv
         }
     }
 
+    @Override
+    public StreamObserver<Message> invokeBidiStreaming(Endpoint endpoint, Object request, InvokeContext ctx,
+                                                       InvokeCallback callback, long timeoutMs) {
```
Comment (fengjiachun) [inline jraft-core/src/main/java/com/alipay/sofa/jraft/rpc/impl/BoltRpcClient.java:127]: 建议参考原有代码规范，添加 final 关键字

File: jraft-extension/rpc-grpc-impl/src/main/java/com/alipay/sofa/jraft/rpc/impl/GrpcClient.java [hunk 3]
```diff
@@ -111,63 +112,112 @@ public void registerConnectEventListener(final ReplicatorGroup replicatorGroup)
...
+		}
+		catch (final TimeoutException e) {
+			future.cancel(true);
+			throw new InvokeTimeoutException(e);
+		}
+		catch (final Throwable t) {
+			future.cancel(true);
+			throw new RemotingException(t);
+		}
+	}
```
Comment (fengjiachun) [inline jraft-extension/rpc-grpc-impl/src/main/java/com/alipay/sofa/jraft/rpc/impl/GrpcClient.java:139]: 这个格式需要手动复原一下

File: jraft-extension/rpc-grpc-impl/src/main/java/com/alipay/sofa/jraft/rpc/impl/GrpcClient.java [hunk 4]
```diff
@@ -111,63 +112,112 @@ public void registerConnectEventListener(final ReplicatorGroup replicatorGroup)
...
-            public void onCompleted() {
-                // NO-OP
-            }
-        });
+			@Override
+			public void onCompleted() {
+				// NO-OP
+			}
+		});
+	}
```
Comment (fengjiachun) [inline jraft-extension/rpc-grpc-impl/src/main/java/com/alipay/sofa/jraft/rpc/impl/GrpcClient.java:174]: 上面的代码需要手动复原格式

File: jraft-extension/rpc-grpc-impl/src/main/java/com/alipay/sofa/jraft/rpc/impl/GrpcClient.java (hunk 3 above)
Comment (HHoflittlefish777) [reply jraft-extension/rpc-grpc-impl/src/main/java/com/alipay/sofa/jraft/rpc/impl/GrpcClient.java:139]: 这个格式是由mvn clean compile格式化的，需要复原吗？

File: jraft-core/src/main/java/com/alipay/sofa/jraft/rpc/RpcClient.java [hunk 5]
```diff
@@ -107,4 +110,31 @@ default void invokeAsync(final Endpoint endpoint, final Object request, final In
...
+     * Streaming invocation with a callback.
+     *
+     * @param endpoint  target address
+     * @param request   request object
+     * @param callback  invoke callback
+     * @param timeoutMs timeout millisecond
+     * @return request  stream observer.
+     */
+    default StreamObserver<Message> invokeBidiStreaming(final Endpoint endpoint, final Object request, final InvokeCallback callback,
+            final long timeoutMs) {
```
Comment (fengjiachun) [inline jraft-core/src/main/java/com/alipay/sofa/jraft/rpc/RpcClient.java:124]: 这一行好像格式没对齐

File: jraft-core/src/main/java/com/alipay/sofa/jraft/rpc/impl/BoltRpcClient.java [hunk 6]
```diff
@@ -119,6 +123,18 @@ public void invokeAsync(final Endpoint endpoint, final Object request, final Inv
...
+    @Override
+    public StreamObserver<Message> invokeBidiStreaming(final Endpoint endpoint, final Object request, final InvokeContext ctx,
+                                                       final InvokeCallback callback, final long timeoutMs) {
+        try {
+            throw new UnSupportException("UnSupport this method");
+        }
+        catch (final UnSupportException e) {
+            e.printStackTrace();
+        }
+        return null;
```
Comment (fengjiachun) [inline jraft-core/src/main/java/com/alipay/sofa/jraft/rpc/impl/BoltRpcClient.java:135]: 需要抛出异常，不能吞掉

**Comment by fengjiachun** [comment]
代码质量未达预期，先关闭了

**Pull request closed without merging**